nvflare/apis/impl/controller.py

```diff
@@ -169,125 +172,126 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> Tuple[str,
             raise TypeError("fl_ctx must be an instance of FLContext, but got {}".format(type(fl_ctx)))
 
         client_task_to_send = None
-        with self._task_lock:
-            self.logger.debug("self._tasks: {}".format(self._tasks))
-            for task in self._tasks:
-                if task.completion_status is not None:
-                    # this task is finished (and waiting for the monitor to exit it)
-                    continue
+        with self._controller_lock:
+            with self._task_lock:
```

Comment:
Double locking should be avoided if all possible. It can easily cause dead-lock. In this case, I don't see the need for the controller_lock.

Reply:
Thanks for the review, I simplified the logic